src/LanguageServer/Protocol/ExternalAccess/Razor/GetSymbolicInfoHandler.cs

```diff
@@ -0,0 +1,312 @@
+        if (classDeclarationNode is null)
+        {
+            return null;
+        }
+
+        // Get the block syntax directly inside method BuildRenderTree(RenderTreeBuilder builder). This will be the most ancestral (first) block syntax.
+        var blockNode = classDeclarationNode.DescendantNodes().OfType<BlockSyntax>().FirstOrDefault();
+        if (blockNode is null)
+        {
+            return null;
+        }
+
+        var dataFlowAnalysis = semanticModel.AnalyzeDataFlow(blockNode);
+        var writtenInsideSymbols = dataFlowAnalysis.WrittenInside;
+
+        // Using the generated spans as a criterion to traverse through the tree generally returns incomplete results.
+        // Instead, we get all of the identifiers, methods, fields, and properties in the class.
+        // Then we get the identifiers that are within the generated spans.
+        // We then find the methods, fields, and properties that correspond to the identifiers within the generated spans.
+        var identifiersBuilder = ArrayBuilder<IdentifierNameSyntax>.GetInstance();
+        var methodsBuilder = ArrayBuilder<MethodDeclarationSyntax>.GetInstance();
+        var fieldsBuilder = ArrayBuilder<FieldDeclarationSyntax>.GetInstance();
+        var propertiesBuilder = ArrayBuilder<PropertyDeclarationSyntax>.GetInstance();
+        foreach (var node in classDeclarationNode.DescendantNodes())
+        {
+            switch (node.Kind())
+            {
+                case SyntaxKind.IdentifierName:
+                    identifiersBuilder.Add((IdentifierNameSyntax)node);
+                    break;
+                case SyntaxKind.MethodDeclaration:
+                    methodsBuilder.Add((MethodDeclarationSyntax)node);
+                    break;
+                case SyntaxKind.FieldDeclaration:
+                    fieldsBuilder.Add((FieldDeclarationSyntax)node);
+                    break;
+                case SyntaxKind.PropertyDeclaration:
+                    propertiesBuilder.Add((PropertyDeclarationSyntax)node);
+                    break;
+            }
+        }
+
+        var identifiersInClass = identifiersBuilder.ToImmutableAndFree();
+        var methodsInClass = methodsBuilder.ToImmutableAndFree();
+        var fieldsInClass = fieldsBuilder.ToImmutableAndFree();
+        var propertiesInClass = propertiesBuilder.ToImmutableAndFree();
+
+        var identifiersInRange = identifiersInClass.Where(identifier => generatedSpans.Any(span => span.Contains(identifier.Span)))
+                    .Select(identifier => new IdentifierAndSymbol
+                    {
+                        Identifier = identifier,
+                        Symbol = semanticModel.GetSymbolInfo(identifier).Symbol
+                    })
+                    .Where(x => x.Symbol is not null);
```

Comment:
Consider moving this into the above loop, and simply building `identifiersInRange` inside that loop, rather than collecting all identifiers and filtering later.